src/mame/sega/suprloco.cpp

```diff
@@ -464,48 +594,71 @@ static const gfx_layout charlayout =
+	// Audio CPU INTs are caused by the falling edge of the V5 line counter
+	//  (4x per frame) and are auto-acked.
+	///TODO: once pals are dumped, make sure this uses the correct timing!
+	m_audiocpu->set_periodic_int(FUNC(suprloco_state::irq0_line_hold), attotime::from_hz(4 * 60));
+	// Audio CPU NMIs are caused by the main CPU via PPI portC bit7, see below
 
 	i8255_device &ppi(I8255A(config, "ppi"));
 	ppi.out_pb_callback().set(FUNC(suprloco_state::control_w));
 	ppi.tri_pb_callback().set_constant(0);
-	ppi.out_pc_callback().set_output("lamp0").bit(0).invert(); // set by 8255 bit mode when no credits inserted
+	ppi.out_pc_callback().set_output("lamp0").bit(0).invert(); ///TODO: 8255 portc bit 0 is MUTE, same as in system1, not a lamp.
 	ppi.out_pc_callback().append_inputline(m_audiocpu, INPUT_LINE_NMI).bit(7).invert();
 
 	// video hardware
-	screen_device &screen(SCREEN(config, "screen", SCREEN_TYPE_RASTER));
-	screen.set_refresh_hz(60);
-	screen.set_vblank_time(ATTOSECONDS_IN_USEC(5000));
-	screen.set_size(32*8, 32*8);
-	screen.set_visarea(1*8, 31*8-1, 0*8, 28*8-1);
-	screen.set_screen_update(FUNC(suprloco_state::screen_update));
-	screen.set_palette("palette");
-
-	GFXDECODE(config, m_gfxdecode, "palette", gfx_suprloco);
-	PALETTE(config, "palette", FUNC(suprloco_state::palette), 512 + 256);
+	SCREEN(config, m_screen, SCREEN_TYPE_RASTER);
+	///TODO: use correct timing once PALs are dumped
+	m_screen->set_refresh_hz(60);
+	m_screen->set_vblank_time(ATTOSECONDS_IN_USEC(5000));
+	m_screen->set_size(32*8, 32*8);
+	m_screen->set_visarea(1*8, 31*8-1, 0*8, 28*8-1);
+	m_screen->set_screen_update(FUNC(suprloco_state::screen_update));
+	m_screen->set_palette(m_palette);
+
+	GFXDECODE(config, m_gfxdecode, m_palette, gfx_suprloco);
+	PALETTE(config, m_palette, FUNC(suprloco_state::palette), 1024);
 
 	// sound hardware
 	SPEAKER(config, "mono").front_center();
 
-	SN76496(config, "sn1", 4'000'000).add_route(ALL_OUTPUTS, "mono", 1.0);
-
-	SN76496(config, "sn2", 2'000'000).add_route(ALL_OUTPUTS, "mono", 1.0);
+	///TODO: sn1 goes into the pin 9 input of sn2, which is poorly documented for mixing; instead, we just duplicate the second filter here for now
+	FILTER_RC(config, m_filter_1).set_rc(filter_rc_device::HIGHPASS, RES_K(27), 0, 0, CAP_U(10)); //CR high-pass R5+C78+VR1(2kohm)
+	m_filter_1->add_route(ALL_OUTPUTS, "mono", 0.707);
+	FILTER_RC(config, m_filter_1a).set_rc(filter_rc_device::HIGHPASS, RES_K(22), 0, 0, CAP_U(10)); //CR high-pass R2+C72
+	m_filter_1a->add_route(ALL_OUTPUTS, m_filter_1, 1.0);
+	SN76489A(config, "sn1", MASTER_CLOCK/5).add_route(ALL_OUTPUTS, m_filter_1a, 1.0); // see above re: /5 divider
+	FILTER_RC(config, m_filter_2).set_rc(filter_rc_device::HIGHPASS, RES_K(27), 0, 0, CAP_U(10)); //CR high-pass R5+C78+VR1(2kohm)
+	m_filter_2->add_route(ALL_OUTPUTS, "mono", 0.707);
+	SN76489A(config, "sn2", MASTER_CLOCK/10).add_route(ALL_OUTPUTS, m_filter_2, 1.0); // see above re: /5 divider (itself divided by 2)
```

Comment:
It’s an RC filter irrespective of which way around the components are.

Reply:
I'm not quite sure what you wanted me to change here, but I changed the comments slightly.